flang/runtime/misc-intrinsic.cpp

```diff
@@ -67,7 +68,13 @@ void RTDECL(Rename)(const Descriptor &path1, const Descriptor &path2,
   // We simply call rename(2) from POSIX
   int result = rename(pathSrc, pathDst);
   if (status) {
-    StoreIntToDescriptor(status, result, terminator);
+    // When an error has happened,
+    int errorCode = 0; // Assume success
+    if (result < 0) {
```

Comment:
nit: The Windows documentation just says "a non-zero value"
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename

Reply:
Good catch!  I have changed that.  Windows effectively reports -1 (which is in line with the docs :-)), but I have made it such that it works in any case.